azkaban_cli/azkaban.py

```diff
@@ -409,3 +433,124 @@ def fetch_projects(self):
+            project,
+            group, 
+            permission_options
+        )
+
+        self.__catch_response_error_ignore_empty(response, AddPermissionError)
+        
+        logging.info('Group [%s] add with permission in the Project [%s] successfully' % (group, project))
+
+    def remove_permission(self, project, group):
+        """Remove permission command, intended to make the request to Azkaban and treat the response properly.
+
+        This method receives the project name and the group name and execute 
+        request to remove a group permission from the project and evaluate the response.
+
+        :param project: Project name on Azkaban
+        :type project: str
+
+        :param group: Group name on Azkaban
+        :type project: str
+
+        """
+
+        self.__check_if_logged()
+
+        response = api.remove_permission_request(
+            self.__session,
+            self.__host,
+            self.__session_id,
+            project,
+            group 
+        )
+
+        self.__catch_response_error_ignore_empty(response, RemovePermissionError)
+        
+        logging.info('Group [%s] permission removed from the Project [%s] successfully' % (group, project))
+
+
+    def change_permission(self, project, group, permission_options):
+        """Change permission command, intended to make the request to Azkaban and treat the response properly.
+
+        This method receives the project name, the group name, and the permission options and execute 
+        request to change a existing group permission in a project and evaluate the response.
+
+        :param project: Project name on Azkaban
+        :type project: str
+
+        :param group: Group name on Azkaban
+        :type project: str
+
+        :param permission_options: The group permissions in the project on Azkaban
+        :type project: Dictionary
+        """
+
+        self.__check_if_logged()
+
+        permission_options = self.__check_group_permissions(permission_options)
+
+        response = api.change_permission_request(
+            self.__session,
+            self.__host,
+            self.__session_id,
+            project,
+            group, 
+            permission_options
+        )
+
+        self.__catch_response_error_ignore_empty(response, ChangePermissionError)
+        
+        logging.info('Group [%s] received new permissions in the Project [%s] successfully' % (group, project))
+
+    def __check_group_permissions(self, permission_options):
+        #validate the dictionary options
+        if('admin' not in permission_options): permission_options['admin'] = False
+        if('write' not in permission_options): permission_options['write'] = False
+        if('read' not in permission_options): permission_options['read'] = False
+        if('execute' not in permission_options): permission_options['execute'] = False
+        if('schedule' not in permission_options): permission_options['schedule'] = False
```

Comment:
This could be changed to:
```
from collections import defaultdict

[...]

filled_permission_options = defaultdict(lambda: False)
filled_permission_options.update(permission_options)
```
As well.

Reply:
> This could be changed to:
> 
> ```
> from collections import defaultdict
> 
> [...]
> 
> filled_permission_options = defaultdict(lambda: False)
> filled_permission_options.update(permission_options)
> ```
> 
> As well.

Actually, forget this. You need the keys explicit in the dict to pass as arg to the request.